toolkit/tools/internal/packagerepo/repocloner/rpmrepocloner/rpmrepocloner.go

```diff
@@ -181,6 +194,21 @@ func (r *RpmRepoCloner) Initialize(destinationDir, tmpDir, workerTar, existingRp
 		return
 	}
 
+	r.disableMarinerRepos = disableMarinerRepos
+	if r.disableMarinerRepos {
```

Comment:
I think we may be able to simplify by modifying `initializeRepoDefinitions`. That function already gathers the user-provided and built-in repos into one file. All we'd need to do, is skip adding the built-in repos whenever `r.disableMarinerRepos` is true. In other words: skip the `for _, originalRepoFilePath := range existingRepoFiles` loop.

This way we wouldn't need to parse the repo files at all, I believe.

**NOTE**: the `repoDefinitions` variable already contains the built-in repo files `fetcher.repo` and `local.repo`, which are required for the chroot to function properly with the local caches.

Reply:
I think we'd have to remove the existing repo files instead of skipping the copy to `allrepos.repo`. This approach makes sense if there's a guarantee that the existing repo files consist only of PMC repos

Reply:
also, the `getRepoList` function is only used to get the repo IDs that are going to be disabled

Reply:
I updated my code to follow this logic